Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Smoothing/smoothing_evaluation.h

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2018 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0+
+//
+//
+// Author(s)     : Konstantinos Katrioplas (konst.katrioplas@gmail.com)
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_SMOOTHING_EVALUATION_H
+#define CGAL_POLYGON_MESH_PROCESSING_SMOOTHING_EVALUATION_H
+
+#include <CGAL/license/Polygon_mesh_processing/meshing_hole_filling.h>
+
+#include <vector>
+#include <set>
+#include <fstream>
+#include <string>
+#include <boost/graph/graph_traits.hpp>
+#include <boost/foreach.hpp>
+#include <boost/property_map/property_map.hpp>
+#include <CGAL/Polygon_mesh_processing/measure.h>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+namespace internal {
+
+template<typename PolygonMesh, typename GeomTraits>
+class Quality_evaluator
+{
+  typedef typename GeomTraits::Point_3 Point;
+  typedef typename GeomTraits::Vector_3 Vector;
+  typedef typename GeomTraits::Line_3 Line;
+  typedef typename boost::graph_traits<PolygonMesh>::halfedge_descriptor halfedge_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::face_descriptor face_descriptor;
+  typedef typename boost::property_map<PolygonMesh, CGAL::vertex_point_t>::type VertexPointMap;
+
+public:
+  Quality_evaluator(PolygonMesh& pmesh) : mesh_(pmesh)
+  {
+    std::size_t number_of_triangles = faces(mesh_).size();
+    angles_.reserve(number_of_triangles * 3);
+    areas_.reserve(number_of_triangles);
+    aspect_ratios_.reserve(number_of_triangles);
+    vpmap_ = get(CGAL::vertex_point, mesh_);
+  }
+
+  void gather_angles()
+  {
+    double rad_to_deg = 180. / CGAL_PI;
+
+    BOOST_FOREACH(halfedge_descriptor hi, halfedges(mesh_))
+    {
+      Point a = get(vpmap_, source(hi, mesh_));
+      Point b = get(vpmap_, target(hi, mesh_));
+      Point c = get(vpmap_, target(next(hi, mesh_), mesh_));
+      Vector ba(b, a);
+      Vector bc(b, c);
+      double cos_angle = (ba * bc)
+        / std::sqrt(ba.squared_length() * bc.squared_length());
+      angles_.push_back(std::acos(cos_angle) * rad_to_deg);
+    }
+    std::cout<<"angles_ size= "<<angles_.size()<<std::endl;
```

Comment:
here and everywhere else, the verbose code should be protected by a macro, for example
```
#ifdef CGAL_PMP_SMOOTHING_VERBOSE
 std::cout << "blabla" << std::endl;
#endif
```

Reply:
I just saw this macro is already being used elsewhere in this PR, so it is the right one!